### ISSUE TITLE
Separate traits for querying for a stream of responses vs. a single response

### ISSUE THREAD

**Issue by divergentdave** [hickory-dns/hickory-dns#2571]
The following TODO suggests changing `DnsHandle` to return one response instead of a stream of responses.

https://github.com/hickory-dns/hickory-dns/blob/64953998f8350d3cc8ebf125316e9e734182a7c5/crates/recursor/src/recursor_dns_handle.rs#L384-L385

A stream of responses is needed for mDNS and DNS-SD use cases, but using the same trait for `RecursorDnsHandle` and `DnssecDnsHandle`, which produce exactly one response, adds complexity to their implementation and use.

See also #383 and #1478.

**Comment by djc** [comment]
Aren't there also other cases in normal resolving where we want to yield multiple results for a single query (like with EDNS or just over TCP)?

**Comment by bluejekyll** [comment]
Yeah, this is a generic issue, I'm not sure what we should really do here. We also need a stream for AXFR I believe. I'm definitely open to ideas here.
